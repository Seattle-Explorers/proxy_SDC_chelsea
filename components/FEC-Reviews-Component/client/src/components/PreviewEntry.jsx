import React from 'react';
import styled from 'styled-components';
import moment from 'moment';

// const BasicInfoStyle = styled.div`
//   display: flex;
//   flex-direction: column;
// `
// const AvatarStyle = styled.img`
//   width: 50px;
//   height: 50px;
//   border-radius: 50%;
//   flex-direction: column;
// `
// const DateStyle = styled.span`
//   color: #B3B3B3;
// `
// const BodyStyle = styled.div`
//   flex-direction: row;
// `
const Wrapper = styled.div`
  margin: 25px 50px 25px 50px;  
`
const TopStyle = styled.div`
  display: flex;
`
const AvatarStyle = styled.img`
  width: 50px;
  height: 50px;
  border-radius: 50%;
  flex-direction: column;
`
const TopText = styled.div`
  flex-direction: row
`
const Date = styled.span`
  color: #B3B3B3;
`
class PreviewEntry extends React.Component {
  constructor(props) {
    super(props)

    this.state = {
      readMore: null,
    }

    this.handleClick = this.handleClick.bind(this);
  }

  componentDidMount() {
    const { review } = this.props;
    const { body } = review;

    if (review.body.length > 180) {
      this.setState({
        readMore: true,
        display: body.slice(0,180)
      })
    } else {
      this.setState({
        readMore: false,
        display: body
      })
    }
  }

  handleClick() {
    const { review } = this.props;
    const { body } = review;

    this.setState({
      readMore: false,
      display: body
    })
  }

  render() {
    const { review, idx } = this.props;
    const imageURL = `https://dteamdp.s3-us-west-2.amazonaws.com/pug${review.dp}.jpg`
    const date = moment(review.date).format('MMMM YYYY');

    if(this.state.readMore) {
      return(
        <Wrapper>
          <TopStyle>
            <AvatarStyle src={imageURL} />
            <TopText>
              {review.reviewer_name}
              <br></br>
              <Date>{date}</Date>
            </TopText>
          </TopStyle>
          {this.state.display}
          <a href='#' onClick={this.handleClick}>...Read More</a>
        </Wrapper>
      )
    } else {
      return(
        <Wrapper>
          <TopStyle>
            <AvatarStyle src={imageURL} />
            <TopText>
              {review.reviewer_name}
              <br></br>
              <Date>{date}</Date>
            </TopText>
          </TopStyle>
          {this.state.display}
        </Wrapper>
      )
    }
  }
}

export default PreviewEntry;