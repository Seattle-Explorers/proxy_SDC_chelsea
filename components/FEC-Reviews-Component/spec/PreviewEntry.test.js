import 'jsdom-global/register';
import React from 'react';
import renderer from 'react-test-renderer';
import 'jest-enzyme';
import 'jest-styled-components';
import { mount } from 'enzyme';

import PreviewEntry from '../client/src/components/PreviewEntry';

describe('Preview Entry Components', () => {
  const review = {
    reviewer_name: 'Maureen',
    dp: '11',
    date: '2014-09-19T19:33:37.736Z',
    body: new Array(185).join('1'),
  };

  it('should render', () => {
    const component = <PreviewEntry review={review} />;

    const tree = renderer.create(component).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('should render an image from S3', () => {
    const url = 'https://dteamdp.s3-us-west-2.amazonaws.com/pug11.jpg';

    const wrapper = mount(<PreviewEntry review={review} />);
    expect(wrapper.find('img').prop('src')).toEqual(url);
  });

  it('review body display should be variable', () => {
    const wrapper = mount(<PreviewEntry review={review} />);

    const string180 = review.body.slice(0, 180);

    expect(wrapper.state('display')).toEqual(string180);

    wrapper.find('a').simulate('click');

    expect(wrapper.state('display')).toEqual(review.body);
  });
});
