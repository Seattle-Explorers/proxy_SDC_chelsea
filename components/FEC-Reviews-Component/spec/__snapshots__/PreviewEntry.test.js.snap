// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Preview Entry Components should render 1`] = `
.c0 {
  margin: 25px 50px 25px 50px;
}

.c1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
}

.c2 {
  width: 50px;
  height: 50px;
  border-radius: 50%;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
}

.c3 {
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
}

.c4 {
  color: #B3B3B3;
}

<div
  className="c0"
>
  <div
    className="c1"
  >
    <img
      className="c2"
      src="https://dteamdp.s3-us-west-2.amazonaws.com/pug11.jpg"
    />
    <div
      className="c3"
    >
      Maureen
      <br />
      <span
        className="c4"
      >
        September 2014
      </span>
    </div>
  </div>
  111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111
  <a
    href="#"
    onClick={[Function]}
  >
    ...Read More
  </a>
</div>
`;
